### PR TITLE
feature/cloudflare-typeform

### LINKED ISSUE #6: Add Cloudflare Access protection for Vanity Typeform
**Feature**
Add Cloudflare Access protected page with Typeform for generating the vanity links

**Background**
Cloudflare Access allows us to protect specific domains, subdomains or pages to only be allowed access if a user has authenticated themselves through an identity provider. Our configuration of Cloudflare Access allows for authenticated G Suite users to be able to access these protected _ACM_ Organization resources. 

For the specific issue, we seek to add a layer of security around the typeform that allows for the generation of vanity links. The current setup allows for anyone that has access to the raw link to be able to bypass _ACM_ and generate links under the `acmutd.co` domain. With the implementation of this protected page, we will be able to ensure that any user that seeks to fill out the form has signed in with their G Suite account. 

The typeform being used for this supports two hidden fields --> `email` and `name`. Hidden fields can be added to the end of the typeform url to pass along information without directly having to enter it as a specific question. More details about them can be found in the [typeform docs](https://help.typeform.com/hc/en-us/articles/360029114712-How-to-use-Hidden-Fields).

**Development Task List**
 - [x] Create view for displaying the Typeform. Should be a full page component.
 - [x] Retrieve CF_Authorization token
 - [x] Decode and ensure that it is a valid JWT
 - [x] Extract email from JWT
 - [x] Extract first name and last name from email
 - [x] ACM email format = `firstname.lastname@acmutd.co`
 - [x] Update the `tfLink` parameter to include those two fields
 - [x] Example `tfLink` = `https://linktothetypeform.com#email=harsha.srikara@acmutd.co&name=Harsha%20Srikara`
 - [x] Update the typeform url to use an environment variable

**Deployment Task List**
 - [ ] Use the _ACM_ Azure account to deploy this (Free with [Azure SWA](https://azure.microsoft.com/en-us/services/app-service/static/) and provides all the same features as premium Heroku)
 - [x] Setup DNS as `vanity.acmutd.co`
 - [x] Protect the `vanity` subdomain with Cloudflare Access

**Additional context**
The purpose of this is to serve as a simple example of using Cloudflare Access to protect _ACM_ Organization specific resources. Successful completion of this issue will set the foundation for future additions that make use of this feature. Regarding the specific issue itself, this is only the first iteration and we will continue to adapt and improve its functionality over time.

### PR DESCRIPTION
**Pull request by harshasrikara**
Resolves #6

### CODE REVIEW

**Comment by harshasrikara** [comment]
More [Information](https://docs.google.com/document/d/1EwdRjJWhr6ktnQcRLBeXfjXw5lp9SGbE7r8VJ-I7mLg/edit?usp=sharing)

File: src/App.tsx [hunk 1]
```diff
@@ -14,15 +15,19 @@ function App() {
       <BrowserRouter>
         <BrowserView>
           <Switch>
-            <Route path="/" component={Homepage} exact />
```
Comment (ReshmiCode) [inline src/App.tsx:17]: should this be undone? or the /vanity path removed

File: src/index.tsx [hunk 2]
```diff
@@ -7,8 +7,6 @@ import config from "./config/auth0_config";
 import { Provider } from "react-redux";
 import store from "./store/store";
 
-console.log(config);
```
Comment (ReshmiCode) [inline src/index.tsx:10]: the best kind of debugging hehe

Comment (ReshmiCode) [reply src/index.tsx:10]: ooP also the CI build failed so yes fix that too

**Review by ReshmiCode — APPROVED** [review]

**Pull request merged**